### DISCUSSION TITLE
[Question] 编辑器右上角按钮菜单，如何控制按钮的enable disable

### DISCUSSION THREAD

**Discussion by 767746649zyy** [opensumi/core#2144] [Q&A] [answered]
**描述你的问题（Describe you question here）**
我的需求是编辑器右上角有两个按钮菜单，一个开始执行，一个暂停执行，点击开始执行后，开始执行按钮置灰变成不可用状态，当点击暂停执行后开始执行按钮变成可用状态。相关ui和代码如下

![050798d74721704fcfdc075d6a293b9](https://user-images.githubusercontent.com/23164065/209913951-8571db57-b109-455b-babd-e7f32b71507c.png)

![image](https://user-images.githubusercontent.com/23164065/209913934-536dd285-76fb-47aa-9be3-0b02ccc2d53e.png)
问题：如何在execute执行函数里改变两个按钮的可用状态

**Comment by erha19** [comment, marked as answer]
@767746649zyy 如果想动态控制 EditorTitle 区域的菜单是否可用状态，可用通过注册时追加 `enabledWhen` 属性，与在 `Command` 中使用 `isEnabled` 区别的是，`when` 类型的参数会动态响应条件变化，但需要你自主注册条件，如，你可以直接通过使用 `IContextKeyService` 去注册自定义 ContextKey，伪代码如下：

```ts
@Autowired(IContextKeyService)
private readonly contextKeyService: IContextKeyService;

const key = contextKeyService.createKey<boolean>('customAction.isRunning', false);

// 更新
key.set(true);
```

注册伪代码：
```ts
  menuRegistry.registerMenuItem(MenuId.DebugBreakpointsContext, {
     ...
     enableWhen: `... && custom.isRunning`
  });
```

**Reply by 767746649zyy** [reply]
明白了，非常感谢
